jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/types/Asn1Choice.java

```diff
@@ -0,0 +1,28 @@
+package us.dot.its.jpo.asn.runtime.types;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import us.dot.its.jpo.asn.runtime.serialization.SerializationUtil;
+
+@Slf4j
+public abstract class Asn1Choice implements Asn1Type {

```

Comment:
question: this seems to be a duplicate of Asn1Sequence, and both appear to just be wrapping a boolean value. Are two classes like this really required or can we consolidate?

Reply:
> question: this seems to be a duplicate of Asn1Sequence, and both appear to just be wrapping a boolean value. Are two classes like this really required or can we consolidate?

Yes, SEQUENCE and CHOICE are different asn.1 types.  A sequence is like a C-struct or Java POJO.  A CHOICE is like a C Union and has no real Java equivalent, but it can only have one choice selected at a time. They are serialized differently.  This library has special deserializers/serializers for CHOICE types.

Reply:
ah I see, can we add some class level documentation to clarify what these are representing to avoid confusion in the future?

Reply:
> ah I see, can we add some class level documentation to clarify what these are representing to avoid confusion in the future?

Yes, added javadocs to all base classes